src/hotspot/share/nmt/memTracker.hpp

```diff
@@ -166,6 +167,36 @@ class MemTracker : AllStatic {
     }
   }
 
+  static inline MemoryFileTracker::MemoryFile* register_device(const char* descriptive_name) {
+    assert_post_init();
+    if (!enabled()) return nullptr;
```

Comment:
Could we push `assert_post_init()` into `enabled()` ?

Reply:
That's a discussion that should take place in its own PR.

Reply:
I see 16 instances of the same patter:

```
    assert_post_init();
    if (!enabled()) return
```

`in memTracker.hpp` so it's local and isolated to `MemTracker` class. I didn't think this would be controversial/big deal to warrant its own PR?

Reply:
Maybe not, but this PR is already very large :). Changing this can be done in a separate PR and is not directly related to these changes, so it should be done in its separate PR.